### ISSUE TITLE
consider marking some dependencies as optional

### ISSUE THREAD

**Issue by jacopo-exact** [fractal-analytics-platform/fractal-tasks-core#164]
right now this package requires to download ~3GB of packages in order to be installed.
consider marking some / all of the heavy weight dependencies as optional, so that one could install the pacakge and still enjoy limited functionality without installing the full dependencies.

of course this requires that the single tasks inform the user about potential missing dependencies and how to install them with the extra syntax.

**Comment by tcompa** [comment]
The obvious candidate for being an extra is cellpose (a quick test suggests that without it we are down to 1.5G), but the question is back to @jacopo-exact: how would it work for the task collection on the fractal-server's side? I guess we can go either way:
1. We add the handling of package extras in fractal-server.
2. We tell the user: `cellpose` is not available in standard environments, please provide an environment you created yourself. IMHO this option goes against the `tasks-core` package.. then it would be better to split this repo into a couple of packages, but that also has quite a few disadvantages.

TL;DR
My take is that we currently keep all the dependencies, to avoid complicating things in the task-collection side and to guarantee a standard set of core tasks available to everyone.

**Comment by tcompa** [comment]
Testing of the architecture kept going, even with this heavier dependencies.
Closing this for now.
